feat:refactoring and created responsive design for footer

SvitlanaYaroslavicka1 opened the pull request:
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/ita-social-projects/EventsExpress-Frontend/projects/1)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

SvitlanaYaroslavicka1 committed:
refactored and created responsive design

SvitlanaYaroslavicka1 committed:
style(created own file for mixins): created own file for mixins

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -22,35 +23,34 @@ const socialLinks = [
+          <img src={LogoWhite} alt="Logo" className="footer__logo" />
+        </Link>
+        <div className="footer__links_to_pages_container">
+          {links.map(link => (
```
It probably missed during latest PRs, but I'd like to ask you to fix it in this PR.
This `map` call makes from links array, which is constant and will not be changed. So all components, which are created from these links are always same.
Could you please extract this `map` call from this place and move it outside of component function into some constant variable

SvitlanaYaroslavicka1 replied:
DOne

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -22,35 +23,34 @@ const socialLinks = [
-          </Link>
-        ))}
+    <footer className="footer__container">
+      <div className="footer__links_and_logo_container">
```
Please don't use prepositions in classnames. In this case there is no problem to create wrapper div only for Link and specify simplified name

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -22,35 +23,34 @@ const socialLinks = [
+        <Link to="/home">
+          <img src={LogoWhite} alt="Logo" className="footer__logo" />
+        </Link>
+        <div className="footer__links_to_pages_container">
```
Same as above

SvitlanaYaroslavicka1 replied:
DOne

andrii-kuzmichov commented on src/components/footer/footer.js:
```diff
@@ -1,15 +1,16 @@
+import { IoMdSend } from "react-icons/io";
+import LogoWhite from "../../assets/icons/LogoWhite.png";
 import "./footer.scss";
 import constants from "../../constants/Footer";
```
Rename

SvitlanaYaroslavicka1 replied:
DOne

andrii-kuzmichov commented on src/components/footer/footer.js:
```diff
@@ -1,15 +1,16 @@
+const { SUBSCRIBE_US, CONTACT_US, PHONE_NUMBER, PASTE_YOUR_EMAIL } = constants;
 
 const links = [
   { id: 0, name: "Home", path: "/home" },
```
Maybe use specific `id` instead of index of array

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.js:
```diff
@@ -22,35 +23,34 @@ const socialLinks = [
+        </Link>
+        <div className="footer__links_to_pages_container">
+          {links.map(link => (
+            <Link key={link.id} to={link.path} className="footer__link">
```
Can you drop id property from mapper and use path for key?

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.js:
```diff
@@ -22,35 +23,34 @@ const socialLinks = [
-        <div className="social-links__icons">
+        <div className="footer__social_links_icons">
           {socialLinks.map(link => (
             <Link key={link.id} to={link.link} className="nav-link social-link">
```
the same as above

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,102 @@
-    display: flex;
-    flex-direction: row;
+.footer__subscribe {
+  @include flex-direction(column);
```
What do this mixin?

SvitlanaYaroslavicka1 replied:
DOne

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,102 @@
-justify-content: center;
-    gap: 5px;
+.footer__link {
+  font-size: 1.5em;
```
Can you use px for font-size?

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,102 @@
-    margin-left: auto;
+.footer__input {
+  background-color: $mainSearchGrey;
+  font-family: $mainFont;
+  font-style: normal;
+  font-weight: 700;
```
create mixin and use it

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,102 @@
+  .footer__social_links h4,
+  .footer__subscribe h4,
+  .footer__link {
+    font-size: 1em;
```
Do we need to use `em`?

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,102 @@
+  .footer__links_to_pages_container {
+    margin-top: 5px;
+    margin-left: 5px;
+    font-size: 0.7em;
```
The same as above

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/styles/_mixins.scss:
```diff
@@ -0,0 +1,11 @@
+  justify-content: $justify-c;
+}
+
+@mixin flex-direction($flex-d) {
+  display: flex;
+  flex-direction: $flex-d;
+}
```
useless mixin

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/styles/main.scss:
```diff
@@ -2,4 +2,5 @@
-@import "./buttons";
\ No newline at end of file
+@import "./buttons";
+@import"./mixins.scss";
```
Drop `.sccs` from import and add new line

SvitlanaYaroslavicka1 replied:
Done

SvitlanaYaroslavicka1 committed:
refactor(fixed mistakes): fixed mistakes

changed mapper, added constants, delete em, rename div

SvitlanaYaroslavicka1 committed:
fix(footer.js footer.js): rename path

andrii-kuzmichov commented on src/components/footer/footer.scss:
```diff
@@ -1,118 +1,108 @@
+  .footer__links {
+    margin-top: 5px;
+    margin-left: 5px;
+    // font-size: $fontsize-14;
```
clean up

SvitlanaYaroslavicka1 replied:
Done

andrii-kuzmichov commented on src/components/footer/footer.js:
```diff
@@ -1,60 +1,81 @@
-              <i className={link.icon} />
-            </Link>
+        <div className="footer__social_links_icons">
+          {SOCIALLINKS.map(({ link, icon }) => (
```
`SOCIALLINKS` => `SOCIAL_LINKS`

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton commented on src/constants/Footer.js:
```diff
@@ -3,5 +3,23 @@ const FOOTER = {
+  CREATE_EVENT: "Create event",
+  TERMS: "Terms",
+  ABOUT: "About",
+  PATH_HOME: "/home",
```
Can we group some of these constants into few categories
For examples: All PATH_* into PATHS, LINK_* into LINKS and so one

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -1,60 +1,81 @@
+          <img src={LogoWhite} alt="Logo" className="footer__logo" />
+        </NavLink>
+        <div className="footer__links">
+          {LiNKS.map(({ path, name }) => (
```
That's was not done as I asked before, I mean regarding extracting this `LINKS.map` outside to component

SvitlanaYaroslavicka1 replied:
Done

SvitlanaYaroslavicka1 replied:
I hope I understood what you mean.

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -1,60 +1,81 @@
-              <i className={link.icon} />
-            </Link>
+        <div className="footer__social_links_icons">
+          {SOCIALLINKS.map(({ link, icon }) => (
```
As well as for LINKS.map, please move it outside of component

SvitlanaYaroslavicka1 replied:
Done

SvitlanaYaroslavicka1 committed:
refactor(footer.js  footer. scss): delete coment, rename mapper

SvitlanaYaroslavicka1 committed:
refactor(footer-mapper.js footer.js footerconstants.js): move mappers, group constants

group some of these constants into few categories, move mapper outside of component

VadymKharyton commented on src/components/footer/footer.js:
```diff
@@ -1,62 +1,38 @@
+        <NavLink to={HOME}>
+          <img src={LogoWhite} alt="Logo" className="footer__logo" />
+        </NavLink>
+        <div className="footer__links">{MapperLink("footer__link_item")}</div>
```
Is there any another usage of MapperLink, that requires us to manually pass className?
If we use it only one time, it doesn't require to pass it inside. Just specify it directly into the specific tag

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton commented on src/components/footer/footer-mapper.js:
```diff
@@ -0,0 +1,52 @@
+  { icon: STYLE_ICON_YOUTUBE, link: LINK_YOUTUBE },
+];
+
+export const MapperLink = classLink => {
```
Please rename it to `NAV_LINKS`

And I meant before to make it as constant variable, not a function.
Something like `const NAV_LINKS = LINKS.map(...)`. 
It will be created one time here, and then we can just to use it component piece inside main component, and it will not be recreated each rerender

SvitlanaYaroslavicka1 replied:
I did it for const SOCIAL_NAV_LINKS =     too

VadymKharyton commented on src/components/footer/footer-mapper.js:
```diff
@@ -0,0 +1,52 @@
+const { STYLE_ICON_FACEBOOK, STYLE_ICON_INSTAGRAM, STYLE_ICON_YOUTUBE } =
+  ALL_STYLE_ICON;
+
+const LiNKS = [
```
Typo in the name, Should be `LINKS`

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton commented on src/components/footer/footer-mapper.js:
```diff
@@ -0,0 +1,52 @@
+  ));
+};
+
+export const MapperSocialLink = () => {
```
Please rename it to `SOCIAL_NAV_LINKS`, and make the same as above

SvitlanaYaroslavicka1 replied:
Done

VadymKharyton reviewed (commented):
There few more comments, which I would like to see resolved. Then it can be approved

SvitlanaYaroslavicka1 committed:
refactor(footer-maper.js): make  constant varibles
